src/ray/rpc/grpc_server.cc

```diff
@@ -61,8 +61,11 @@ void GrpcServer::Shutdown() {
 
 void GrpcServer::Run() {
   uint32_t specified_port = port_;
-  std::string server_address((listen_to_localhost_only_ ? "127.0.0.1:" : "0.0.0.0:") +
-                             std::to_string(port_));
+  const char *ray_ip = std::getenv("RAY_PREFER_IPV6");
+  bool use_ipv6 = ray_ip != nullptr;
+  std::string server_address(
+      (listen_to_localhost_only_ ? "127.0.0.1:" : (use_ipv6 ? "[::]:" : "0.0.0.0:")) +
+      std::to_string(port_));
```

Comment:
Swapping `"127.0.0.1"` -> `"::"` without also binding `"::1"` (loopback) may expose the Dashboard, PDB, or control plane APIs on the host’s external IPv6 interface. Best practice is to bind local-only services to `::1` explicitly, and wildcard services (cluster-in traffic) to `::`.